mlflow/utils/file_utils.py

```diff
@@ -198,7 +199,7 @@ def read_file_lines(parent_path, file_name):
     :return: All lines in the file as an array.
     """
     file_path = os.path.join(parent_path, file_name)
-    with open(file_path, 'r') as f:
+    with codecs.open(file_path, mode='r', encoding="utf-8-sig") as f:
```

Comment:
Sorry if I missed discussion on this elsewhere, but why utf-8-sig instead of utf-8?

Reply:
Saw some concerns online about BOM leading bits to handling encoding. I ran into an error while trying to read tags. 
ref: https://docs.python.org/3/library/codecs.html and https://stackoverflow.com/questions/23154355/python-utf-8-sig-bom-in-the-middle-of-the-file-when-appending-to-the-end and 

however, it seems to work with `utf-8` as well